File: app/src/main/java/com/jingjoeh/bottomnavigationviewexample/MainActivity.java
package com.jingjoeh.bottomnavigationviewexample;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
	TextView textView;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);

		textView = findViewById(R.id.text);

		BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_nav_view);
		bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
			@Override
			public boolean onNavigationItemSelected(@NonNull MenuItem item) {
				switch (item.getItemId()) {
					case R.id.item_recent:
						textView.setText("Recent");
						return true;
					case R.id.item_favorite:
						textView.setText("Favorite");
						return true;
					case R.id.item_nearby:
						textView.setText("Nearby");
						return true;
				}
				return false;
			}
		});
		bottomNavigationView.setSelectedItemId(R.id.item_favorite);

	}
}

File: app/src/main/java/com/jingjoeh/bottomnavigationviewexample/MainActivity.kt
package com.jingjoeh.bottomnavigationviewexample

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        bottomNavigationView.setOnNavigationItemSelectedListener { item ->
            when (item.itemId) {
                R.id.item_recent -> {
                    textView.text = "Recent"
                    return@setOnNavigationItemSelectedListener true
                }
                R.id.item_favorite -> {
                    textView.text = "Favorite"
                    return@setOnNavigationItemSelectedListener true
                }
                R.id.item_nearby -> {
                    textView.text = "Nearby"
                    return@setOnNavigationItemSelectedListener true
                }
            }
            false
        }

        bottomNavigationView.setOnNavigationItemReselectedListener {
            Toast.makeText(this, "${it.title} is Reselected", Toast.LENGTH_SHORT).show()
        }

        // select item
        bottomNavigationView.selectedItemId = R.id.item_favorite
    }

}
